### PR TITLE
Add parameter type declarations to magic methods

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32179
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

**Review by Tobion — APPROVED** [review]

File: src/Symfony/Component/Validator/Constraints/File.php [hunk 1]
```diff
@@ -91,7 +91,10 @@ public function __get($option)
         return parent::__get($option);
     }
 
-    public function __isset($option)
+    /**
+     * @return bool
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Constraints/File.php]: should be removed: the parent has it already (same below)

Comment (derrabus) [reply src/Symfony/Component/Validator/Constraints/File.php]: With below you mean `Symfony\Component\VarDumper\Cloner\Data`? That class does not have a parent.

File: src/Symfony/Component/VarDumper/Cloner/Data.php [hunk 2]
```diff
@@ -137,7 +137,10 @@ public function __get($key)
         return null;
     }
 
-    public function __isset($key)
+    /**
+     * @return bool
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Cloner/Data.php]: See #33300

**Comment by derrabus** [comment]
I have resolved the conflict. PR is ready again.

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @derrabus.

**Pull request merged**